tests/test_builtin_casters.cpp

```diff
@@ -147,6 +147,17 @@ TEST_SUBMODULE(builtin_casters, m) {
     // test_reference_wrapper
     m.def("refwrap_builtin", [](std::reference_wrapper<int> p) { return 10 * p.get(); });
     m.def("refwrap_usertype", [](std::reference_wrapper<UserType> p) { return p.get().value(); });
+    m.def("refwrap_usertype_const", [](std::reference_wrapper<const UserType> p) { return p.get().value(); });
+
+    m.def("refwrap_lvalue", []() -> std::reference_wrapper<UserType> {
+        static UserType x(1);
+        return std::ref(x);
+    });
+    m.def("refwrap_lvalue_const", []() -> std::reference_wrapper<const UserType> {
+        static UserType x(1);
+        return std::cref(x);
+    });
```

Comment:
This is related to `type_caster<std::reference_wrapper<T>>::cast`, which isn't touched by this PR? Why add these tests?

Reply:
Because I saw no test returning a `std::reference_wrapper` at all and I didn't want things to break. I think that it's reasonable to add.

Reply:
True, but it's unclear that it's not related to the functional changes in this PR.

Reply:
(or at least, it was to me)

Reply:
I'd say keep the tests. They aren't hurting anything.